src/spec.c

```diff
@@ -81,6 +81,55 @@ static void setMemoryInfo(RedisModuleCtx *ctx) {
   RedisModule_FreeServerInfo(ctx, info);
 }
 
+#define MILLISECOND_IN_ONE_SECOND 1000
+#define NANOSECOND_IN_ONE_MILLISECOND 1000000
+
+static inline timespec timespecFromMilliseconds(int64_t totalMilliseconds) {
+  timespec result = {.tv_sec = 0, .tv_nsec = 0};
+  if (totalMilliseconds > 0) {
+    result.tv_sec = totalMilliseconds / MILLISECOND_IN_ONE_SECOND;
+    result.tv_nsec = (totalMilliseconds % MILLISECOND_IN_ONE_SECOND) * NANOSECOND_IN_ONE_MILLISECOND;
+  }
+  return result;
+}
+
+static inline t_expirationTimePoint getDocExpirationTime(RedisModuleCtx* ctx, RedisModuleString *keystr) {
+  timespec zero = {.tv_sec = 0, .tv_nsec = 0};
+  RedisModuleCallReply *rep = RedisModule_Call(ctx, "PEXPIRETIME", "s", keystr);
+  if (rep == NULL || RedisModule_CallReplyType(rep) != REDISMODULE_REPLY_INTEGER) {
+    size_t length = 0;
+    const char* error = rep ? RedisModule_CallReplyStringPtr(rep, &length) : "unknown error";
+    RedisModule_Log(ctx, "warning", "Error calling PEXPIRETIME, type: %d, error(%d): %s", rep ? RedisModule_CallReplyType(rep) : REDISMODULE_REPLY_UNKNOWN, errno, error);
+    return zero;
```

Comment:
Its more about the API for RedisModule_Call.
The documentation for it is:
`On success a RedisModuleCallReply object is returned, otherwise NULL is returned and errno is set to the following values:

    EBADF: wrong format specifier.
    EINVAL: wrong command arity.
    ENOENT: command does not exist.
    EPERM: operation in Cluster instance with key in non local slot.
    EROFS: operation in Cluster instance when a write command is sent in a readonly state.
    ENETDOWN: operation in Cluster instance when cluster is down.
    ENOTSUP: No ACL user for the specified module context
    EACCES: Command cannot be executed, according to ACL rules
    ENOSPC: Write or deny-oom command is not allowed
    ESPIPE: Command not allowed on script mode
`

Reply:
not sure if any of the options are possible in this case (at least not the error reply case), but if it is let's free the reply

Reply:
Okay